ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/Genesis/Tests/DensityDisconnect.hs

```diff
@@ -278,7 +295,12 @@ evolveBranches EvolvingPeers {k, sgen, peers = initialPeers} =
               }
           (loeFrag, suffixes) = sharedCandidatePrefix curChain candidates
           disconnect = fst (densityDisconnect sgen k states suffixes loeFrag)
-      either (pure . second (result loeFrag)) step (updatePeers sgen target disconnect next)
+          updated
+            | AF.anchorToSlotNo (AF.anchor loeFrag) > AF.headSlot (candidate (value (honest ps)))
+            = Left (Finished, ps)
```

Comment:
Why must the test be terminated in this case?

I would expect this to not be possible unless the honest chain is disconnected.

Reply:
because the adversary forks off the last block of the honest chain (maybe this only happens here because the honest candidate wasn't completed by the random fragment-extender?)

Reply:
> because the adversary forks off the last block of the honest chain

I follow until there. I think you mean that the adversary forks off the last block of the candidate fragment of the honest peer. How does that become problematic then?

Reply:
now that you say it like that, I realize that there should still be an intersection...

I only saw that my in my test, the LoE frag was anchored in the adversarial fragment, and assumed that the reason was that there was no actual "fork" (i.e. the honest chain had no extra blocks).
but `sharedCandidatePrefix` should still use the last honest block as the anchor, I guess.